torchaudio/functional.py

```diff
@@ -2314,3 +2321,93 @@ def vad(
     res = waveform[:, pos - samplesLen_ns + flushedLen_ns:]
     # unpack batch
     return res.view(shape[:-1] + res.shape[-1:])
+
+
+def add_background_noise(
+    waveform: Tensor,
+    noise: Tensor,
+    snr: float,
+    normalize: bool = True
+)-> Tensor:
+    r"""
+    Augment noise to waveform, with specified SNR value to generate degraded waveform.
```

Comment:
Based on this description, it is not clear to me what this function would do beyond "addition". What use case do you have in mind for such a user-facing function? Would this be better as a private function?

Reply:
Naively, this function would consume the color of the noise that a user wants to add?